ext/json/json_encoder.c

```diff
@@ -31,6 +31,15 @@
 
 static const char digits[] = "0123456789abcdef";
 
+static zend_always_inline bool php_json_check_stack_limit(void)
+{
+#ifdef ZEND_CHECK_STACK_LIMIT
+	return zend_call_stack_overflowed(EG(stack_limit));
+#else
+	return false;
+#endif
+}
+
```

Comment:
Wouldn't make sense to make this not json specific as it might be useful for serialization as well possibly?

Reply:
Maybe this should be moved to the `Zend/zend_call_stack.h` header file, I think it would make sense there. What do you think @arnaud-lb ?

Reply:
Agree, it would make sense to move this function to `Zend/zend_call_stack.h`

Reply:
It's a bit annoying to do so as it would require us to pull in zend_globals.h to get access to `EG` fields, which pulls in a whole bunch more..